test/integration/loginControllerTests.ts

```diff
@@ -162,4 +163,102 @@ describe('LoginControllerTests', () => {
         assert.equal(body.handled, false, 'Incorrect handled value')
         expect(body.csrf, 'Missing csrfToken value').length.above(0)
     })
+
+    it('An incorrectly configured client secret should not cause a redirect loop in the SPA', async () => {
```

Comment:
I would name the test `An incorrectly configured client secret should return a 400 response` so that it focuses on what our code does.